setup.py

```diff
@@ -17,12 +17,15 @@
 from distutils.sysconfig import get_config_var, get_python_inc
 from setuptools import setup, Extension
 
+CUDA_LIB = os.path.join(os.environ.get('CUDA_PATH'), 'lib64')
+CUDA_INC = os.path.join(os.environ.get('CUDA_PATH'), 'include')
```

Comment:
What's the rationale for `CUDA_PATH` vs. `CUDA_HOME`? I see mostly `CUDA_HOME` in the wild, but I haven't found any references or definitive guides about what is preferred.

Also, are these paths meant to be optional? If `CUDA_PATH` is not set, `os.environ.get("CUDA_PATH")` will return `None` and the call to `os.path.join` will fail.

Reply:
I think `CUDA_PATH` is the one used in the wild. IIRC `CUDA_HOME` was used in certain Windows env.

Reply:
Ooh, interesting! Here's what I've seen in the past:
- [Arrow appears to use _mostly_ CUDA_HOME](https://github.com/apache/arrow/search?q=CUDA_HOME)
- PyTorch has mixed opinions:
  - [setup.py lists CUDA_HOME (Linux/OS X); CUDA_PATH (Windows)](https://github.com/pytorch/pytorch/blob/dc5cda0cca429a3080f63e7b30b3e87d18df8601/setup.py#L144-L145)
  - [torch/utils/cpp_extension.py uses CUDA_HOME (first) or CUDA_PATH (second)](https://github.com/pytorch/pytorch/blob/a9f44b22c018a59192eea20d358f0742ad48c571/torch/utils/cpp_extension.py#L60).
- [PyCUDA uses CUDA_PATH but only for Windows, and does not use CUDA_HOME at all](https://github.com/inducer/pycuda/blob/ca345f2f59eec4c60832dad5187ef75b396315b3/pycuda/driver.py)
- [Numba prefers CUDA_HOME but falls back to CUDA_PATH on "Windows CUDA installation without Anaconda"](https://github.com/numba/numba/blob/39271156a52c58ca18b15aebcb1c85e4a07e49ed/numba/cuda/cuda_paths.py#L121-L131)
- [nvcc feedstock on conda-forge sets both CUDA_HOME and CUDA_PATH on all platforms but has more references to CUDA_HOME](https://github.com/conda-forge/nvcc-feedstock/search?q=CUDA_HOME)

Reply:
Yeah we've done the same research in the past but I can't find the reference right away, but another data point is if we use the search bar on the CTK documentation (https://docs.nvidia.com/cuda/index.html) to search `CUDA_HOME` we find nothing, while there're three entries for `CUDA_PATH`.